### PR TITLE
Add runtime error codes

### PR DESCRIPTION
**Pull request by danielgranhao**

### CODE REVIEW

File: src/errors.rs [hunk 1]
```diff
@@ -13,6 +13,20 @@
 //! }
 //! ```
 
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum RuntimeErrorCode {
+    RemoteServiceUnavailable,
+    GenericError,
```
Comment (gcomte) [inline src/errors.rs:21]: Not sure whether I like it to propose that one could use an Error that doesn't say anything about its nature. Why is this required?

Comment (danielgranhao) [reply src/errors.rs:21]: We might want to throw an error which we know is a RuntimeError, but we don't have a code for it.

Comment (danielgranhao) [reply src/errors.rs:21]: For example, [syncing the bdk wallet](https://docs.rs/bdk/0.25.0/bdk/wallet/struct.Wallet.html#method.sync) can return a [generic BDK error](https://docs.rs/bdk/0.25.0/bdk/enum.Error.html#variant.Generic). What should we do in that case?

File: src/errors.rs [hunk 2]
```diff
@@ -13,6 +13,20 @@
 //! }
 //! ```
 
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum RuntimeErrorCode {
+    RemoteServiceUnavailable,
```
[the remark this answers is not in the document]
Comment (danielgranhao) [reply src/errors.rs:20]: Added here eaae0bc

File: src/errors.rs (hunk 1 above)
Comment (gcomte) [reply src/errors.rs:22]: Yeah, well in that case we can only forward the inaccurateness, I agree.

Could we at least have a string to add more detail to the error, similar to what the generic BDK error does, or is that not possible because of how our error handling works (resp. because of UNIFFI binding limitations)?

Comment (danielgranhao) [reply src/errors.rs:22]: When throwing this error code we still have the `message` field of in the `RuntimeError` so I think we don't need to add further details as part of the code itself.

Comment (gcomte) [reply src/errors.rs:22]: Oh okay, thanks for elaborating!

**Review by gcomte — APPROVED** [review]
